neat-form-core/src/main/java/com/nerdstone/neatformcore/form/json/JsonFormBuilder.kt

```diff
@@ -94,16 +98,33 @@ class JsonFormBuilder() : FormBuilder {
     internal fun parseJsonForm() {
         form = when {
             this::formString.isInitialized && formString.isNotNull() -> parseJson<NForm>(formString)
-            fileSource.isNotNull() -> parseJson<NForm>(
-                AssetFile.readAssetFileAsString(context, fileSource!!)
-            )
+            fileSource.isNotNull() -> {
+
+                val bundleName = LanguageHelper.getBundleNameFromFileSource(fileSource!!)
+                val rawJsonStringTemplate = AssetFile.readAssetFileAsString(context, fileSource!!)
+
+                parseJson<NForm>(
+                    parseTemplate(bundleName, Locale.getDefault(), rawJsonStringTemplate)
+                )
+
+            }
             else -> null
         }
     }
 
+    private fun parseTemplate(bundleName: String, locale: Locale, template: String) =
+        if (bundleName.endsWith("i18n", ignoreCase = true))
```

Comment:
@ellykits I don't think we need to force the user to name his file to include the string `i18n`. I'd recommend we revert to the previous implementation so that it can be named exactly the same as the corresponding json form as a convention.